### PR TITLE
Added option to insert a space between number and unit

### PR DESCRIPTION
**Pull request by glegrain**

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -31,6 +31,7 @@ Format the given value in bytes into a string. If the value is negative, it is k
 | decimalPlaces | `number`&#124;`null` | Maximum number of decimal places to include in output. Default value to `2`. |
 | fixedDecimals | `boolean`&#124;`null` | Whether to always display the maximum number of decimal places. Default value to `false` |
 | thousandsSeparator | `string`&#124;`null` | Example of values: `' '`, `','` and `.`... Default value to `' '`. |
+| spaceBeforeUnit | `boolean`&#124;`null`| Whether to add a space before unit. Default value to `false` |
```
Comment (dougwilson) [inline Readme.md]: Please insert this row in alphabetical order like the others :)

File: index.js [hunk 2]
```diff
@@ -47,6 +47,7 @@ var parseRegExp = /^((-|\+)?(\d+(?:\.\d+)?)) *(kb|mb|gb|tb|b)?$/i;
  *  decimalPlaces: [number]
  *  fixedDecimals: [boolean]
  *  thousandsSeparator: [string]
+ *  spaceBeforeUnit: [boolean]
```
Comment (dougwilson) [inline index.js]: Please insert this row in alphabetical order like the others :)

File: index.js [hunk 3]
```diff
@@ -75,6 +76,7 @@ function bytes(value, options) {
  * @param {number} [options.decimalPlaces=2]
  * @param {number} [options.fixedDecimals=false]
  * @param {string} [options.thousandsSeparator=]
+ * @param {string} [options.spaceBeforeUnit=false]
```
Comment (dougwilson) [inline index.js]: Please insert this row in alphabetical order like the others :)

File: index.js [hunk 4]
```diff
@@ -112,6 +115,9 @@ function format(value, options) {
     str = str.replace(formatThousandsRegExp, thousandsSeparator);
   }
 
+  if (spaceBeforeUnit) {
+    str = str + ' ';
+  }
```
Comment (dougwilson) [inline index.js]: Space after this bracket.

File: test/bytes.js [hunk 5]
```diff
@@ -21,5 +21,7 @@ describe('Test constructor', function(){
   it('Should convert a number into a string with options', function(){
     // This function is testes more accurately in another test suite
     assert.equal(bytes(1000, {thousandsSeparator: ' '}), '1 000B');
+    // Test space before unit option
+    assert.equal(bytes(1024, {spaceBeforeUnit: true}), '1 kB');
```
Comment (dougwilson) [inline test/bytes.js]: Please add this as an individual test (generally only one assert per `it`) and it should describe what it is testing (that the spaceBeforeUnit works).

**Comment by dougwilson** [comment]
Thanks! Looks good, just a few nits.

**Comment by glegrain** [comment]
Thank you for your comments. I pushed the changes you suggested. Let me know if I should change anything else.

**Comment by dougwilson** [comment]
Thanks, @glegrain, merging now! I just realized an eve better option would be `unitSeparator`, which would let people use any character, not just a space or no space! I'm simply going to rewrite your commits for this, and you'll still be the author, unless you object to that.

**Comment by glegrain** [comment]
Awesome! Indeed, that's an even better solution that I haven't thought of.

**Pull request closed without merging**